### DISCUSSION TITLE
gs-quant for programmers who are beginners in trading

### DISCUSSION THREAD

**Discussion by bibhabasumohapatra** [goldmansachs/gs-quant#177] [Q&A]
Is there tutorial for those who know programming and stats and interested in quantitative modelling of markets?

**Comment by Raminder2799** [comment]
There is an investment and portfolio construction with the implementation of python specialization by EDHEC online you can look at.
(in my opinion, it starts out quite basic but the parts 3 and 4 can get challenging at times)

**Comment by heykav** [comment]
There's actually a real, structured tutorial series shipped inside this repo itself: `gs_quant/documentation` has a numbered notebook progression (`00_data` → `01_markets` → `02_pricing_and_risk` → `03_portfolios` → `04_backtesting` → `05_factor_models` → ... → `12_scenarios`), with `Index.ipynb`/`Contents.ipynb` as the entry point. That's the most direct path from "knows programming and stats" to "quantitative modelling of markets" using this exact library, since it walks through gs-quant's own object model in order (instruments/pricing before backtesting, backtesting before factor models, etc.) rather than assuming you already know the API.

Two things worth knowing before you start, so you don't hit a wall halfway through:

1. **Full API access needs institutional credentials.** A lot of gs-quant (live pricing, risk, real market data) requires a GS client id/secret, which is only issued to institutional clients — see the README. The notebooks and much of the historical/analytics tooling still work without it, but if a beginner tutorial online assumes live data access and yours doesn't authenticate, that's why — not something broken on your end.
2. **General quant-market-modelling background, separate from this library:** if you want the underlying theory (not gs-quant-specific), the standard on-ramps most practitioners actually use are Hull's *Options, Futures, and Other Derivatives* for derivatives pricing fundamentals, and QuantLib's own documentation/examples if you want to see the same concepts (Black-Scholes, binomial trees, term structure) implemented open-source and inspectable line by line. Having that theory background first makes gs-quant's own notebooks click a lot faster, since the library assumes you already know what a Greek or an implied vol surface *is* and is teaching you its own API for computing them.
